client/task_runner.go

```diff
@@ -93,6 +93,12 @@ type TaskRunner struct {
 	// Must acquire persistLock when accessing
 	taskDirBuilt bool
 
+	// createdResources are all the resources created by the task driver
+	// across all attempts to start the task.
+	//
+	// Must acquire persistLock when accessing
```

Comment:
Since AllocRunner calls TaskRunner.SaveState concurrently with TaskRunner.Run executing accessing any field serialized by TaskRunner.SaveState needs synchronizing with a lock.

Simply reading createdResources from within `TaskRunner.run()` is safe from races and wouldn't require a lock, but I don't think we ever do that. For the most part I try to guard even reads in those cases as it's much easier to figure out locking when it's done consistently.

(I considered getting around this by making SaveState go through a chan that `TaskRunner.run()` receives on and builds a state blob without having to worry about concurrent writes. However that would block `AllocRunner.SaveState` for the duration of long operations like pulling images which seems unacceptable.)